internal/cmd/login/login.go

```diff
@@ -115,25 +117,17 @@ func (lc *Cmd) Run() error {
 		return err
 	}
 
-	if err := lc.readUser(); err != nil {
+	if err := lc.checkLoginFlow(); err != nil {
 		return err
 	}
 
-	if err := lc.readPassword(); err != nil {
-		return err
-	}
-
-	if len(lc.user) == 0 || len(lc.password) == 0 {
-		return errors.New("username/password should not be empty")
-	}
-
 	options := &auth.Options{
 		ClientID:     lc.clientID,
 		ClientSecret: lc.clientSecret,
 		IssuerURL:    info.TokenIssuerURL,
 		SSLDisabled:  lc.sslDisabled,
 	}
-	authStrategy, options, err := lc.authBuilder(options)
+	authStrategy, options, err := lc.authBuilder(options, lc.clientCredentialsFlow)
```

Comment:
suggestion - not mandatory, but maybe would be cleaner:
Have the flow as a field in the `Options` e.g. `Options.AuthFlow` where AuthFlow is one of
```golang
type AuthenticationFlow uint8
const (
	PasswordGrantFlow AuthenticationFlow = iota
	ClientCredentialsFlow
)
```
Then iterate over all registered AuthenticationProvider (currently AuthenticationStrategy) and have each AuthenticationProvider will have  `func(AuthenticationProvider) SupportsFlow(AuthenticationFlow) bool`.
You'd have to keep a registry of all authentication providers(strategies) then and maybe switch from `--client-credentials` flag to `--auth-strategy=client_credentials` and then the `AuthenticationFlow` would be:
```golang
type AuthenticationFlow string
const (
	PasswordGrant AuthenticationFlow = "password_grant"
	ClientCredentials AuthenticationFlow = "client_credentials"
        Basic AuthenticationFlow = "basic" // maybe, provide an option to make calls with basic?
)
```